compiler/rustc_middle/src/ty/fold.rs

```diff
@@ -74,7 +76,7 @@ pub trait TypeFoldable<'tcx>: fmt::Debug + Clone {
     }
 
     fn has_type_flags(&self, flags: TypeFlags) -> bool {
-        self.visit_with(&mut HasTypeFlagsVisitor { flags })
+        self.visit_with(&mut HasTypeFlagsVisitor { flags }) == ControlFlow::Break(())
```

Comment:
Yeah, it might make sense to introduce a newtype here in the future, so this can be `visit_with(...).break_value() == Some(FoundFlags)`.

Not part of this PR though